src/main/java/org/broadinstitute/barclay/help/BashTabCompletionDoclet.java

```diff
@@ -0,0 +1,356 @@
+
+        callerScriptOptionsMap.put("callerScriptPrefixLegalArgs", callerScriptPrefixLegalArgs);
+        callerScriptOptionsMap.put("callerScriptPrefixArgValueTypes", callerScriptPrefixArgValueTypes);
+        callerScriptOptionsMap.put("callerScriptPrefixMutexArgs", callerScriptPrefixMutexArgs);
+        callerScriptOptionsMap.put("callerScriptPrefixAliasArgs", callerScriptPrefixAliasArgs);
+        callerScriptOptionsMap.put("callerScriptPrefixMinOccurrences", callerScriptPrefixMinOccurrences);
+        callerScriptOptionsMap.put("callerScriptPrefixMaxOccurrences", callerScriptPrefixMaxOccurrences);
+        if ( hasCallerScriptPrefixArgs ) {
+            callerScriptOptionsMap.put("hasCallerScriptPrefixArgs", "true");
+        }
+        else {
+            callerScriptOptionsMap.put("hasCallerScriptPrefixArgs", "false");
+        }
+
+        callerScriptOptionsMap.put("callerScriptPostfixLegalArgs", callerScriptPostfixLegalArgs);
+        callerScriptOptionsMap.put("callerScriptPostfixArgValueTypes", callerScriptPostfixArgValueTypes);
+        callerScriptOptionsMap.put("callerScriptPostfixMutexArgs", callerScriptPostfixMutexArgs);
+        callerScriptOptionsMap.put("callerScriptPostfixAliasArgs", callerScriptPostfixAliasArgs);
+        callerScriptOptionsMap.put("callerScriptPostfixMinOccurrences", callerScriptPostfixMinOccurrences);
+        callerScriptOptionsMap.put("callerScriptPostfixMaxOccurrences", callerScriptPostfixMaxOccurrences);
+        if ( hasCallerScriptPostfixArgs ) {
+            callerScriptOptionsMap.put("hasCallerScriptPostfixArgs", "true");
+        }
+        else {
+            callerScriptOptionsMap.put("hasCallerScriptPostfixArgs", "false");
+        }
+
+        rootMap.put("callerScriptOptions", callerScriptOptionsMap);
+
+        // Get or create a template
+        final Template template = cfg.getTemplate(getIndexTemplateName());
+
+        // Create the output file
+        final File indexFile = new File(getDestinationDir(),
+                getIndexBaseFileName() + "." + getIndexFileExtension()
+        );
+
+        // Run the template and merge in the data
+        try (final FileOutputStream fileOutStream = new FileOutputStream(indexFile);
+             final OutputStreamWriter outWriter = new OutputStreamWriter(fileOutStream)) {
+            template.process(rootMap, outWriter);
+        } catch (TemplateException e) {
+            throw new DocException("Freemarker Template Exception during documentation index creation", e);
+        }
+    }
+
+    /**
+     * @return the name of the index template to be used for this doclet
+     */
+    @Override
+    public String getIndexTemplateName() { return "bash-completion.template.ftl"; }
+
+    /**
+     * @return The base filename for the index file associated with this doclet.
+     */
+    @Override
+    public String getIndexBaseFileName() { return callerScriptName + "-completion"; }
```

Comment:
1 - For this case I've been trying to find something to do completion, but because of how bash keys off of the first argument, I'm not sure this will work without hooking into the completion for the `java` invocation itself.  If I can find a way to plug into it, then it should not be very hard to make this work.

2 - I added in the "-completion" notation so that it would be clear from the filename that the file is for tab completion.  To be clear, the name of the resulting script doesn't actually matter for this functionality.  To enable completion you source the script and your shell then has the information it needs to complete the options.